rootfs/usr/local/bin/init-terraform

```diff
@@ -4,26 +4,41 @@
 shopt -s nullglob
 TF_BUCKET_REGION=${TF_BUCKET_REGION:-${AWS_REGION}}
 TF_BUCKET_PREFIX=${TF_BUCKET_PREFIX:-$(basename $(pwd))}
+TF_DIR=${@:-1}
 TF_FILE=${TF_FILE:-terraform.tfstate}
-TF_FILES=(*.tf)
-TF_ARGS=("-backend=true" "-backend-config=key=${TF_BUCKET_PREFIX}/${TF_FILE}")
+TF_FILES=(*.tf *.tfvars)
+TF_ARGS=("-backend=true")
+shopt -u nullglob
 
-# Disable color if not running in a TTY
+# Disable color if not running in a TTY (e.g. CI/CD context)
 if [ ! -t 1 ]; then
 	TF_ARGS+=("-no-color")
 fi
 
-[ ! -z ${TF_FILES} ] || (
+# Do not attempt to initialize non-terraform projects
+if [ -z "${TF_FILES}" ] && [ -z "${TF_FROM_MODULE}" ]; then
 	echo "Run $0 from a terraform project directory"
 	exit 1
-)
+fi
 
+# Translate environment variables to terraform arguments
+[ -z "${TF_FROM_MODULE}" ] || TF_ARGS+=("-from-module=${TF_FROM_MODULE}")
+[ -z "${TF_FILE}" ] || TF_ARGS+=("-backend-config=key=${TF_BUCKET_PREFIX}/${TF_FILE}")
 [ -z "${TF_BUCKET}" ] || TF_ARGS+=("-backend-config=bucket=${TF_BUCKET}")
 [ -z "${TF_BUCKET_REGION}" ] || TF_ARGS+=("-backend-config=region=${TF_BUCKET_REGION}")
 [ -z "${TF_DYNAMODB_TABLE}" ] || TF_ARGS+=("-backend-config=dynamodb_table=${TF_DYNAMODB_TABLE}")
 [ -z "${AWS_PROFILE}" ] || TF_ARGS+=("-backend-config=profile=${AWS_PROFILE}")
 [ -z "${AWS_ROLE_ARN}" ] || TF_ARGS+=("-backend-config=role_arn=${AWS_ROLE_ARN}")
 
-s3 mount
+# Mount all filesystems (E.g. `goofys`)
+mount -a
+
+# Initialize terraform state
+terraform init ${TF_ARGS[*]} $*
+
+# Copy the current directory configuration & state to the module
+if [ -d "${TF_DIR}" ] && [ -n "${TF_FILES}" ] && [ -n "${TF_FROM_MODULE}" ]; then
```

Comment:
`TF_FROM_MODULE` not used below?

Reply:
The objective is to only execute this code block when using `TF_FROM_MODULE` and when we have files in the current directory (`TF_FILES`) that we need to "overlay" on top of the existing configuration